test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/ModelStateDictionaryTest.cs

```diff
@@ -733,6 +734,86 @@ public void ModelStateDictionary_NoErrorMessage_ForNonFormatException()
             Assert.Empty(error.ErrorMessage);
         }
 
+        [Fact]
+        public void ModelStateDictionary_ClearEntriesThatMatchWithKey_NonEmptyKey()
+        {
+            var dictionary = new ModelStateDictionary();
+
+            dictionary["Property1"] = new ModelState { ValidationState = ModelValidationState.Valid };
+
+            dictionary["Property2"] = new ModelState { ValidationState = ModelValidationState.Invalid };
```

Comment:
IN all the tests, we should also test for the third state, ModelValidationState.Skipped. For properties marked as skipped should we be making them Unvalidated or should we keep them as skipped?